### DISCUSSION TITLE
How to exclude pattern for a directory except a sub directory

### DISCUSSION THREAD

**Discussion by daniel-horizon** [phpactor/phpactor#2589] [Q&A]
I would like to prevent PHPactor from indexing vendor except for the laravel directory within the vendor, to make indexing faster. Someone knows how I can write this rule in the config as JSON?

**Comment by mamazu** [comment]
This is currently not really possible as there is no hirarchy in the include / exclude process. We already have an issue for that: https://github.com/phpactor/phpactor/issues/2485

**Comment by daniel-horizon** [comment]
That's unfortunate! Not ideal but I currently managed to work around it like so

`
    "indexer.stub_paths": [
        "\/var\/\/www\/\/html\/\/vendor\/laravel",
        "\/var\/\/www\/\/html\/\/vendor\/horizon"
    ],`

**Comment by mamazu** [comment]
That's a good workaround. I might also use it until the other feature is implemented.

**Comment by mamazu** [comment]
It's implemented now.

**Comment by daniel-horizon** [comment]
Hi, it's not very clear how this can work.

I have something like this in my configuration

```
"indexer.include_patterns": ["/**/*.php", "/vendor/laravel/**/*"],
  "indexer.exclude_patterns": [
    "/vendor/**/*",
  ],
 ```
  
 Expecting for all vendor to be excluded except for the files in the /vendor/laravel directory but still nothing more is included.

**Reply by mamazu** [reply]
Yes, this is how it should work.
